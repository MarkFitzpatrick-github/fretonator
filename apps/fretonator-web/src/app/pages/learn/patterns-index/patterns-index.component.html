<div class="patterns__container">
  <div class="patterns__intro">
    <h1 class="patterns__title">Learn the 7 scale patterns on the guitar</h1>
    <h1 class="patterns__subtitle">The 3 note per string method</h1>
  </div>
<hr class="patterns__divider" />
</div>



<div class="patterns__container">

  Ionian pattern (root of major scale)
  G Ionion (major)
</div>

<app-fretboard [fretMap]="gIonianFretMap" [stringNamesAreCaseSensitive]="true"></app-fretboard>

<div class="patterns__container">

  Dorian pattern (2nd of major scale)
  A Dorian
</div>

<app-fretboard [fretMap]="aDorianFretMap" [stringNamesAreCaseSensitive]="true"></app-fretboard>

<div class="patterns__container">

  Phrygian pattern (3rd of major scale)
  B Phrygian
</div>

<app-fretboard [fretMap]="bPhrygianFretMap" [stringNamesAreCaseSensitive]="true"></app-fretboard>
