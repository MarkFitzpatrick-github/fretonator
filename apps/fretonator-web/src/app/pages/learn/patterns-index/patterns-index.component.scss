@import '../../../../styles/vars';
@import '../../../../styles/mixins';
@import '../../../../styles/typography';

.patterns__container {
  @include content_wrapper;
  max-width: $fretonator_max_width;
}

.patterns__intro {
  max-width: 698px;
  margin-bottom: pxToRem($grid-unit * 6);
  margin-top: pxToRem($grid-unit * 6);
}

.patterns__title {
  @include font_large;
  margin-bottom: pxToRem($grid-unit * 2);
}

.patterns__subtitle {
  @include font_medium;
}

.patterns__divider {
  @include divider();
}
