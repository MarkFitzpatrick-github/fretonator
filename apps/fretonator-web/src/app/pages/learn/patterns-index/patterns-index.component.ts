import { Component, EventEmitter, OnInit, Output } from '@angular/core';
import { PatternFretMaps, PatternModeSelectors } from '../../../util/constants';
import { Mode } from '../../../util/types';

@Component({
  selector: 'app-patterns-index',
  templateUrl: './patterns-index.component.html',
  styleUrls: ['./patterns-index.component.scss']
})
export class PatternsIndexComponent implements OnInit {
  @Output() expandFretboard = new EventEmitter<MouseEvent>();
  modeSelectors = PatternModeSelectors;
  selectedMode = Mode.ionian;
  selectedFretMap = PatternFretMaps.ionian;
  modesRequiringExpansion = [
    Mode.mixolydian,
    Mode.aolian,
    Mode.locrian
  ];

  constructor() {
  }

  ngOnInit(): void {
  }

  setPattern(mode: Mode) {
    this.selectedMode = mode;
    this.selectedFretMap = PatternFretMaps[mode];

    if (this.modesRequiringExpansion.indexOf(mode) > -1) {
      this.expandFretboardHere();
    }
  }

  expandFretboardHere() {
    //JIM I JUST CANNOT WORK OUT WHERE TO PUT THIS DIRECTIVE
    // IS IT ON THE INSTANCE OF FRETBOARD ON PATTERNS-INDEX?
    // IS IN IN THE FRETBOARD HTML?
    this.expandFretboard.emit();
  }
}
