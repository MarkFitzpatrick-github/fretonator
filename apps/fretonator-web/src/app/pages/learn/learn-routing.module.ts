import { NgModule } from '@angular/core';
import { Route, RouterModule, Routes } from '@angular/router';
import { LearnIndexComponent } from './learn-index/learn-index.component';
import { PatternsIndexComponent } from './patterns-index/patterns-index.component';
import { ScrollToGuard } from '../../common/scroll-to/scroll-to.guard';

const learnHub: Route = {
  path: '',
  pathMatch: 'full',
  canActivate: [
    ScrollToGuard
  ],
  component: LearnIndexComponent,
};

const patterns: Route = {
  path: 'patterns',
  pathMatch: 'full',
  canActivate: [
    ScrollToGuard
  ],
  component: PatternsIndexComponent,
};

const routes: Routes = [learnHub, patterns];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule]
})
export class LearnRoutingModule { }
