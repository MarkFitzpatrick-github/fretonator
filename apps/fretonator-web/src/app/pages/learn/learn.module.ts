import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { LearnRoutingModule } from './learn-routing.module';
import { LearnIndexComponent } from './learn-index/learn-index.component';
import { StripeBannerModule } from '../../common/stripe-banner/stripe-banner.module';
import { PatternsIndexComponent } from './patterns-index/patterns-index.component';
import { FretboardModule } from '../../common/fretonator/fretboard/fretboard.module';


@NgModule({
  declarations: [
    LearnIndexComponent,
    PatternsIndexComponent
  ],
  imports: [
    CommonModule,
    LearnRoutingModule,
    StripeBannerModule,
    FretboardModule
  ],
  exports: [
    LearnIndexComponent,
    PatternsIndexComponent
  ]
})
export class LearnModule {
}
