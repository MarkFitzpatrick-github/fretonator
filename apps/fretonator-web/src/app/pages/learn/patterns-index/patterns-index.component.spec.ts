import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { PatternsIndexComponent } from './patterns-index.component';
import { Component, DebugElement } from '@angular/core';
import { LearnModule } from '../learn.module';
import { By } from '@angular/platform-browser';
import { RouterTestingModule } from '@angular/router/testing';

describe('PatternsIndexComponent', () => {
  const selectors = {
    chip: By.css('.chip')
  };
  const classNames = {
    activeChip: 'chip--active'
  };

  @Component({
    selector: 'app-patterns-index-spec',
    template: `
      <app-patterns-index></app-patterns-index>
    `
  })
  class PatternsIndexComponentSpec {
  }

  let component: PatternsIndexComponentSpec;
  let fixture: ComponentFixture<PatternsIndexComponentSpec>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [PatternsIndexComponentSpec],
      imports: [
        LearnModule,
        RouterTestingModule
      ]
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(PatternsIndexComponentSpec);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  describe('setPattern,', () => {
    let chip: DebugElement;

    beforeEach(() => {
      chip = fixture.debugElement.queryAll(selectors.chip)[1];
      chip.nativeElement.click();
      fixture.detectChanges();
    });

    it('should activate the clicked chip', () => {
      expect(chip.classes[classNames.activeChip]).toBeTruthy();
    });
  });
});
