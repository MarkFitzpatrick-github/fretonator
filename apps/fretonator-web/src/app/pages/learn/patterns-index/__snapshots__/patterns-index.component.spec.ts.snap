// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatternsIndexComponent should create 1`] = `PatternsIndexComponentSpec {}`;
